runner/runners/queue.go

```diff
@@ -0,0 +1,111 @@
+package runners
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/scootdev/scoot/runner"
+	"github.com/scootdev/scoot/runner/execer"
+	"github.com/scootdev/scoot/snapshot"
+)
+
+const QueueFullMsg = "No resources available. Please try later."
+const RequestIsNotDone = "Run %s is not done, please Abort it first."
+
+// commandAndID is a command waiting to run in the queue and the ID we've assigned
+type commandAndID struct {
+	cmd *runner.Command
+	id  runner.RunId
+}
+
+func NewQueueRunner(exec execer.Execer, filer snapshot.Filer, outputCreator runner.OutputCreator, capacity int) runner.Runner {
+	statuses := NewStatuses()
+	invoker := NewInvoker(exec, filer, outputCreator)
+	controller := &QueueController{statuses: statuses, invoker: invoker, capacity: capacity}
+	return NewControllerAndStatuserRunner(controller, statuses)
+}
+
+// TODO(dbentley): QueueController and SingleController are so similar...
```

Comment:
Suggest Removing Comment:  These two controllers are semantically different right?  Forcing an abstraction on them to save code, will result in code that is harder to maintain / modify down the line